common/lib/capa/capa/capa_problem.py

```diff
@@ -808,7 +822,13 @@ def _extract_html(self, problemtree):  # private
         # otherwise, render children recursively, and copy over attributes
         tree = etree.Element(problemtree.tag)
         for item in problemtree:
-            item_xhtml = self._extract_html(item)
+            item_xhtml = self._extract_html(item, question_id)
+
+            if item.tag == 'question':
+                item_xhtml.set('class', 'question')
+                item_xhtml.set('id', 'question-{}'.format(question_id))
```

Comment:
If you look at the rendered HTML, you will see that solution, explanation, and other fields are doing this concatenation as well. If you can make the format for question similar, that would be nice.